src/librbd/internal.cc

```diff
@@ -1772,6 +1772,7 @@ bool compare_by_name(const child_info_t& c1, const child_info_t& c2)
     r = dest->state->open(false);
     if (r < 0) {
       lderr(cct) << "failed to read newly created header" << dendl;
+      delete dest;
```

Comment:
:-1:  -- ```ImageState::open``` will automatically delete the ```ImageCtx``` upon failure [1]

[1] https://github.com/ceph/ceph/blob/master/src/librbd/ImageState.cc#L254